data/org.freedesktop.portal.Notification.xml

```diff
@@ -106,21 +180,361 @@
 
           * ``label`` (``s``)
 
-            User-visible label for the button. Mandatory.
+            User-visible label for the button.
+            If this property is missing the action may not be displayed.
 
           * ``action`` (``s``)
 
             Name of an action that is exported by the application. The action
-            will be activated when the user clicks on the button. Mandatory.
+            will be activated when the user clicks on the button.
+            If this property is missing the action may not be displayed.
+
+          * ``target`` (``v``)
+
+            Target parameter to send along when activating the action.
+
+          * ``icon`` (``v``)
+
+            A serialized icon to be added to a button. The format for
+            is the same as for the notification ``icon``.
+
+          * ``purpose`` (``s``)
```

Comment:
We should specify explicitly that the `purpose` allows the notification server to completely ignore any specified `label` or `icon`, or any other thing about presentation.

For example, in my call-waiting proposal sometimes `call.accept` will be displayed as two distinct actions: "Hold and Accept" or "End and Accept"